workflows/Snakefile

```diff
@@ -88,46 +92,96 @@ rule gen_cell_simulation:
 
 
 
-# MCMC parameters of mark00
-params_mcmc = dict()
-params_mcmc["seed"] = 42
-params_mcmc["n_trees"] = 1
-params_mcmc["n_cells"] = 2000
-params_mcmc["n_mutations"] = 50
-params_mcmc["strategy"] = """UNIFORM_INCLUDE_ROOT"""
-params_mcmc["fpr"] = 1.24*10**(-6)
-params_mcmc["fnr"] = 0.097
-params_mcmc["na_rate"] = 0.014
-params_mcmc["observe_homozygous"] = "False"
+# choose the optimal move probabilities, by enums
+#optimal_MP_conf = MoveProbConfigOptions.OPTIMAL
+
+
+# Set the Model manually
+#customMcmcConfig = McmcConfigOptions.TEST.value
 
-rule mcmc_mark00:
-    """Run MCMC on mock data."""
+rule make_mcmc_move_prob_config:
+    """Make MCMC move probability config."""
+    params:
+        prune_and_reattach = "{prune_and_reattach}",
+        swap_node_labels = "{swap_node_labels}",
+        swap_subtrees = "{swap_subtrees}"
+    wildcard_constraints:
+        prune_and_reattach = "\d+\.\d+",
+        swap_node_labels = "\d+\.\d+",
+        swap_subtrees = "\d+\.\d+",
+    output:
+        mcmc_move_prob_config = "../data/{experiment}/mcmc/config/MPC_{prune_and_reattach}_{swap_node_labels}_{swap_subtrees}.json",
+    run:
+        # define the move probabilities manually
+        custom_MP_conf = MoveProbConfig(
+            prune_and_reattach=float(params.prune_and_reattach),
+            swap_node_labels=float(params.swap_node_labels),
+            swap_subtrees=float(params.swap_subtrees)
+        )
+
+        with open(output.mcmc_move_prob_config, "w") as f:
+            json.dump(custom_MP_conf.dict(), f)
```

Comment:
Didn't manage to get pydantic objects / Json to be passed to my scripts whilst using simple string matching of snakemake with my filenames.

So my alternative is: have rules that create my configurations pydantic objects using the snakemake wildcards and save those to disk - then to be read in by the later rules. 
- ensures correct typing of my configurations
- allows to use pure string matching of fully informative filenames

Reply:
Agree, this should work :slightly_smiling_face:  Also, I still very much like passing them straight to the rules using `run:` rather than `script:`